internal/exec/terraform_generate_backends.go

```diff
@@ -70,80 +94,130 @@ func ExecuteTerraformGenerateBackends(cliConfig cfg.CliConfiguration, format str
+					if componentType, ok := metadataSection["type"].(string); ok {
+						// Don't include abstract components
+						if componentType == "abstract" {
+							continue
+						}
 					}
 				}
-			}
 
-			// Absolute path to the terraform component
-			backendFilePath := path.Join(
-				cliConfig.BasePath,
-				cliConfig.Components.Terraform.BasePath,
-				terraformComponent,
-				"backend.tf",
-			)
+				// Component backend
+				if backendSection, ok = componentSection["backend"].(map[any]any); !ok {
+					continue
+				}
 
-			if format == "json" {
-				backendFilePath = backendFilePath + ".json"
-			}
+				// Backend type
+				if backendType, ok = componentSection["backend_type"].(string); !ok {
+					continue
+				}
 
-			backendFileAbsolutePath, err := filepath.Abs(backendFilePath)
-			if err != nil {
-				return err
-			}
+				// Find terraform component.
+				// If `component` attribute is present, it's the terraform component.
+				// Otherwise, the YAML component name is the terraform component.
+				terraformComponent := componentName
+				if componentAttribute, ok := componentSection["component"].(string); ok {
+					terraformComponent = componentAttribute
+				}
 
-			// Write the backend config to the file
-			u.PrintMessage(fmt.Sprintf("Writing backend config for the terraform component '%s' to file '%s'", terraformComponent, backendFilePath))
+				// Path to the terraform component
+				terraformComponentPath := path.Join(
+					cliConfig.BasePath,
+					cliConfig.Components.Terraform.BasePath,
+					terraformComponent,
+				)
 
-			if format == "json" {
-				componentBackendConfig := generateComponentBackendConfig(backendType, backendSection)
-				err = u.WriteToFileAsJSON(backendFileAbsolutePath, componentBackendConfig, 0644)
-				if err != nil {
-					return err
+				// Context
+				if varsSection, ok = componentSection["vars"].(map[any]any); !ok {
+					varsSection = map[any]any{}
 				}
-			} else if format == "hcl" {
-				err = u.WriteTerraformBackendConfigToFileAsHcl(backendFileAbsolutePath, backendType, backendSection)
+				context := cfg.GetContextFromVars(varsSection)
+				context.Component = strings.Replace(componentName, "/", "-", -1)
+				context.ComponentPath = terraformComponentPath
+				contextPrefix, err := cfg.GetContextPrefix(stackFileName, context, cliConfig.Stacks.NamePattern, stackFileName)
 				if err != nil {
 					return err
 				}
-			} else {
-				return fmt.Errorf("invalid '--format' argument '%s'. Valid values are 'hcl' (default) and 'json", format)
+
+				var backendFilePath string
+				var backendFileAbsolutePath string
+
+				// Check if `stacks` filter is provided
+				if len(stacks) == 0 ||
+					// `stacks` filter can contain the names of the top-level stack config files:
+					// atmos terraform generate varfiles --stacks=orgs/cp/tenant1/staging/us-east-2,orgs/cp/tenant2/dev/us-east-2
+					u.SliceContainsString(stacks, stackFileName) ||
```

Comment:
Im unfamiliar with the ending syntax here. Is the `||` a typo here? What does it mean?

Reply:
Oh nvm, I see the if statement broken up now. This is a bit confusing syntactically.

Reply:
It is nice you can comment each portion of the if statement I suppose.